sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/batch_deployment.py

```diff
@@ -218,6 +219,9 @@ def _to_rest_object(self, location: str) -> BatchDeploymentData:  # pylint: disa
         )
 
         if is_private_preview_enabled() and self.job_definition:
+            if not self.job_definition.settings and self.compute:
```

Comment:
Why do we need `and self.compute` here?

From what I'm reading then if self.job_definition.settings is None and self.compute is also None, then line 224 will fail.

Reply:
Yes, you are right. Actually, `self.compute` will never be None because it's a required field. I will remove this check and just check that `self.job_definition.settings` is not None